iree/hal/buffer.h

```diff
@@ -106,6 +106,8 @@ enum iree_hal_memory_access_bits_t {
   // within or across operations. The lack of the flag indicates that the access
   // is guaranteed not to alias (ala C's `restrict` keyword).
   IREE_HAL_MEMORY_ACCESS_MAY_ALIAS = 1u << 3,
+  // Memory may have any access.
+  IREE_HAL_MEMORY_ACCESS_ANY = 1u << 4,
```

Comment:
we should probably restrict buffers from being created with this bit - unfortunately that code isn't really centralized so all the creation functions would have to do the masking off of the _ANY bit (iree_hal_heap_buffer_create, etc) - so maybe fine to ignore for now